scripts/zones/Aht_Urhgan_Whitegate/npcs/Ugrihd.lua

```diff
@@ -39,23 +39,27 @@ entity.onTrade = function(player, npc, trade)
 end
 
 entity.onTrigger = function(player, npc)
-    local rank = xi.besieged.getMercenaryRank(player)
-    local badge = xi.besieged.badges[rank]
     local points = player:getCurrency("imperial_standing")
+    local rank   = xi.besieged.getMercenaryRank(player)
+    local badge  = 0
+    if rank > 0 then
+        badge = xi.besieged.badges[rank]
+    end
+
     player:startEvent(150, rank, badge, points, 0, 0, 0, 0, 0, 0)
 end
 
 entity.onEventUpdate = function(player, csid, option)
 end
 
 entity.onEventFinish = function(player, csid, option, npc)
-    if csid == 150 and option < 0x40000000 then
-        local quantity = bit.rshift(option, 0x8)
-        local stacks = math.floor(quantity / 99)
+    if csid == 150 and option < 0x40000000 and option > 255 then
```

Comment:
This has bugged me in the past, that high of an option makes me thing there's just a bit that needs to be checked.

Reply:
Yes, it is exactly that, but I really don't want to start messing arround bits. A refactor can be done at a later time.